### PR TITLE
added faultset datasource

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add fault set data source.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Acceptance Tests

image

### CODE REVIEW

**Comment by shenda1** [comment]
run e2e test

File: examples/data-sources/powerflex_faultset/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# Get faultset details using faultset IDs
+data "powerflex_faultset" "example2" {
+  faultset_ids = ["FaultSet_ID1", "FaultSet_ID2"]
+}
+
+# Get faultset details using faultset names
+data "powerflex_faultset" "example3" {
+  faultset_names = ["FaultSet_Name1", "FaultSet_Name2"]
+}
```
Comment (AnikaAgiwal2711) [inline examples/data-sources/powerflex_faultset/data-source.tf:32]: I believe we'll also have to mention the output block in tmpl file like we did in the documentation enhancements

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,7 @@ module terraform-provider-powerflex
 go 1.20
 
 require (
-	github.com/dell/goscaleio v1.12.1-0.20230911105117-0509b8725a33
+	github.com/dell/goscaleio v1.12.1-0.20231026132856-970807afd308
```
Comment (AnikaAgiwal2711) [inline go.mod:6]: Is this from main ??

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Review by Krunal-Thakkar — DISMISSED** [review]

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

File: go.mod (hunk 2 above)
Comment (shenda1) [reply go.mod]: PR is merged so commit is from main now

**Comment by AnikaAgiwal2711** [comment]
LGTM

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Review by AnikaAgiwal2711 — DISMISSED** [review]

File: powerflex/provider/fault_set_datasource.go [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+									Description:         "Indicates whether the Read RAM Cache is currently temporarily not in use.",
+									MarkdownDescription: "Indicates whether the Read RAM Cache is currently temporarily not in use.",
+									Computed:            true,
+								},
+								"on_vmware": schema.BoolAttribute{
+									Description:         "Presence on VMware.",
+									MarkdownDescription: "Presence on VMware.",
+									Computed:            true,
+								},
+								"faultset_id": schema.StringAttribute{
```
Comment (Krunal-Thakkar) [inline powerflex/provider/fault_set_datasource.go:341]: I think we should change to fault_set_id

**Comment by Krunal-Thakkar** [comment]
Otherwise LGTM

File: powerflex/provider/fault_set_datasource.go (hunk 3 above)
Comment (shenda1) [reply powerflex/provider/fault_set_datasource.go:341]: This is from SDS details. The input params are changed to fault_set_ids. This change will impact SDS datasource.

**Review by Krunal-Thakkar — APPROVED** [review]

**Comment by shenda1** [comment]
run e2e test

**Review by Krunal-Thakkar — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**